portallocator/portallocator.go

```diff
@@ -3,17 +3,36 @@ package portallocator
 import (
 	"errors"
 	"fmt"
+	"github.com/sirupsen/logrus"
 	"net"
 	"sync"
 )
 
-const (
-	// DefaultPortRangeStart indicates the first port in port range
-	DefaultPortRangeStart = 49153
-	// DefaultPortRangeEnd indicates the last port in port range
-	DefaultPortRangeEnd = 65535
+var (
+	// defaultPortRangeStart indicates the first port in port range
+	defaultPortRangeStart = 49153
+	// defaultPortRangeEnd indicates the last port in port range
+	// consistent with default /proc/sys/net/ipv4/ip_local_port_range
+	// upper bound on linux
+	defaultPortRangeEnd = 60999
 )
 
+func sanitizePortRange(start int, end int) (newStart, newEnd int, err error) {
+	if start > defaultPortRangeEnd || end < defaultPortRangeStart || start > end {
```

Comment:
why do we need `start > defaultPortRangeEnd || end < defaultPortRangeStart`

Reply:
this is sanity check. 

say initially, port range is set to 43456-60999, everything works great. Now a (naive) customer comes in and changes published-port-range 80000-90000 ( which is larger (2<<16-1)).  libnetwork would reject this illegal range and continue to use existing/working range for port allocation.